light-client/src/client.rs

```diff
@@ -91,21 +91,22 @@ impl LightClient for ParliaLightClient {
             return Err(Error::ClientFrozen(client_id).into());
         }
 
+        // Create new state and ensure header is valid
+        let trusted_consensus_state = ConsensusState::try_from(any_consensus_state)?;
+
         // Ensure valid validator set
-        self.verify_validator_set(ctx, &client_id, &header)?;
+        header.verify_validator_set(&trusted_consensus_state)?;
```

Comment:
It might be beneficial to include the following process directly within `self.check_header`, which is called in `client_state.check_header_and_update_state`, for better cohesion. What are your thoughts on incorporating this?